subscription.go

```diff
@@ -543,12 +575,16 @@ func (sc *SubscriptionClient) Run() error {
 
 					if closeStatus != -1 && closeStatus < 3000 && closeStatus > 4999 {
 						sc.context.Log(fmt.Sprintf("%s. Retry connecting...", err), "client", GQLInternal)
-						if err = sc.Reset(); err != nil {
+						if err = sc.Run(); err != nil {
 							sc.errorChan <- err
 							return
 						}
 					}
 
+					if isClosedSubscriptionError(err) {
```

Comment:
How will this trigger the exit of the Run function call without closing the context or propagating an error (to `errorChan`)?

Reply:
Currently, there are 2 approaches to exiting the running subscription client:
- Call `client.Close()`.
- Catch any error and return it in the `OnError` callback.
- Or return [ErrSubscriptionStopped](https://github.com/hasura/go-graphql-client/blob/c33bda1416f30f6d769653596c829d7d9a56be4f/subscription.go#L610) directly in the subscription callback.

Is this able to answer your question, or do I miss the point?

Reply:
Oh sorry, my bad. We need to close it also